docs/configuration.md

```diff
@@ -2576,6 +2576,44 @@ newlines.inInterpolation
 - `avoid`: attemps to avoid breaks within the spliced code, regardless of line overflow
 - `oneline`: formats the splice on a single line, or breaks after `${` if overflows
 
+### `newlines.ignoreInSyntax`
+
+The formatter frequently chooses between adding a newline and continuing the
+same line but either prohibiting or heavily discouraging subsequent newlines
+_between_ tokens, to fit the rest of the expression on the same line.
+
+However, in many cases and, for historical reasons, intentionally, newlines
+_within_ tokens have been frequently ignored, leading to "single-line" blocks
+which actually span multiple lines.
+
+This boolean parameter now allows controlling whether to ignore newlines found
+in syntax of strings or other possibly multi-line tokens when newlines are
+otherwise prohibited or undesirable (such as for single-line formatting).
```

Comment:
@tgodzik do you think this explanation makes it a bit clearer?

Reply:
`leading to "single-line" blocks
which actually span multiple lines.` is a bit unclear to me, though I guess the example here would need to be the best explanation we can get